### DISCUSSION TITLE
Precipitation Probabilities

### DISCUSSION THREAD

**Discussion by tmabell** [open-meteo/open-meteo#708] [Ideas]
We've been on the edge of our seats awaiting word on an ETA regarding the addition of Precipitation Probabilities to  API's such as the ECMWF, ICON, BOM & the Canadian GEM.  No one has to emphasize the value and importance of that weather element.  Also, I recently noticed that there is an option in those API's to select Daily Precipitation Probability Max but it doesn't return any data.  If that were a working option and if it were split into day/night (12 hourly) values that would fill a big hole in our needs.

**Comment by patrick-zippenfenig** [comment]
Hi,
this item has been on my ToDo list for the last couple of months (see https://github.com/open-meteo/open-meteo/issues/434). Unfortunately, it got pushed back by other requests and work for customers. Currently, I cannot state any ETA.

**Comment by guidocioni** [comment]
You can still implement this yourself if you take ensemble data, as you can express precipitation probability as the

> (number of ensemble members with rain > 0.1 mm at a given timestep / total number of members) * 100.

This works pretty well also with snow, rain... Only downsides is that you need ensemble data, which is more stuff to download and has lower resolution. But AFAIK there's no way to determine a probability without ensemble models.

I'm doing this in my app pretty easily.

**Comment by patrick-zippenfenig** [comment]
Precipitation probabilities are now calculated for all ensemble models and are used by the regular weather API. See: https://github.com/open-meteo/open-meteo/pull/763

**Comment by tmabell** [comment]
This is welcome news!  Thank you very much!

**Comment by beatagerger** [comment]
uafy-IiJs33x8aVQZQuNCwc-uG-3Ish_JyRNT4RfyK5jWz-YSBSEvCQTcJv9oD6BJSGjA-wP_mfg12h&sa=X&sqi=2&ved=2ahUKEwir0pjupeSFAxWMSFUIHefZCYoQxKsJegQIBxAB&ictx=0)
[Deutsch Russisch](https://www.google.com/search?client=ms-android-motorola-rvo3&sca_esv=e4fcf10d584cbf3e&sca_upv=1&hl=de-AT&gl=at&cs=1&sxsrf=ACQVn0_l5Hgu-VfPi7IPMG5SkuYs7JF0FQ:1714286229870&q=%C3%9Cbersetzer+Deutsch+Russisch&uds=AMwkrPtfdgpRECeqXsCwJLRrC88GyXkJtHnQzgAOthsolr7KzUlF0jvUM00JaD_wbCjAecdsGizv4Mizmp0h6Z0aoxYECC8YqoX65fj4i8JQGg4sV_V2k4bzHfqo2uCmxEgTlG5RlmmYymk8HCMUKwOMFg6XINy_BajFyg-LUb0NdIP1u1AAjt9GqCBazBYVeX-mGzZdyheEFLBDV9I29H0JE-gNmRkpIdEY2O9cQUfGkRGuT8DU3KDe0-GRgUHSeTnZFSt2G-1vOaslrdiq0_C4r4LoFdQAmgtw4RdMIherjArMmdna6prztmgng7R4EQzFj70SLyNx&sa=X&sqi=2&ved=2ahUKEwir0pjupeSFAxWMSFUIHefZCYoQxKsJegQICRAB&ictx=0)
[PONS](https://www.google.com/search?client=ms-android-motorola-rvo3&sca_esv=e4fcf10d584cbf3e&sca_upv=1&hl=de-AT&gl=at&cs=1&sxsrf=ACQVn0_l5Hgu-VfPi7IPMG5SkuYs7JF0FQ:1714286229870&q=PONS+%C3%9Cbersetzer&uds=AMwkrPs5ppTCum7pPx0eKcngOZoTDZz1EC4smCJDcRpLMHtzlwdJuOJukPGghBfbwYHb6GL42vdgxHpcxuxsqWIDKWZjJlU_zBl7TLrlX43p6EwVt09gtUS6D-GyAmu4acgsFfW0uW5-S9Ivfi7UZntEuL2AWJ4Zo0GlK0mwVabJYwTYxqWvQCJFrgme6UxhMwA0jJ1ip-FvKzfjNDdCAk4U9CDUzU31jSkS06ZAj_3wUMY6AEyvtJobGOleW6I4ZPjy3erKTxYHcijKKdCaRzP3OL4OyUxSZoLFt6MJoquyPjoVD01FCH8&sa=X&sqi=2&ved=2ahUKEwir0pjupeSFAxWMSFUIHefZCYoQxKsJegQICxAB&ictx=0)
[Deepl übersetzer](https://www.google.com/search?client=ms-android-motorola-rvo3&sca_esv=e4fcf10d584cbf3e&sca_upv=1&hl=de-AT&gl=at&cs=1&sxsrf=ACQVn0_l5Hgu-VfPi7IPMG5SkuYs7JF0FQ:1714286229870&q=DeepL+%C3%9Cbersetzer&uds=AMwkrPvg5PKm_dNhMKTbEqnEKe3-_82q6L0lZoisKzk85X8WgOk46-MAdHBMQ0jFLcufYmtz98JcM8tBA_chMQGNpaY-TK5sKGH64eAw5CDkOH8EpxMdUUFQfprym2rZb_ols5GHtvm3F7dJExnW_9N6_xS0VKKMPigDXzfHGRd8hRDBGgI0TnNAxqSHYAVcvw2cc462CzHoXYDGKEObYIlPzVEZteQX3Tf_FcpOEQqAumcC6ruVr7tC575z_HVmGElBY_DkmemeviNkBdk5nk3HsMWNxMeBoX4RxXloi2pjQzRcu85JomE&sa=X&sqi=2&ved=2ahUKEwir0pjupeSFAxWMSFUIHefZCYoQxKsJegQIChAB&ictx=0)
[DeepL kostenlos](https://www.google.com/search?client=ms-android-motorola-rvo3&sca_esv=e4fcf10d584cbf3e&sca_upv=1&hl=de-AT&gl=at&cs=1&sxsrf=ACQVn0_l5Hgu-VfPi7IPMG5SkuYs7JF0FQ:1714286229870&q=%C3%9Cbersetzer+DeepL+kostenlos&uds=AMwkrPtEdMXPW1poMpuTxB9R4VbzreoG_B_-24hPllLBKE1v-8IFj1Fm5NXDWhfAckMmrpNNceRqzMGNUoG4CUkEFUeogMrnWLvx7LC1HOsl5uXHEyx_zcEZK5a5pN3qVGi_tXwg7oD_bYQG8kPDZL9grDawEOZPMWUz2Oqvce3MivWrhomvAN29eUszT0dCKBI8nwXoWEw8JeCCB4OlKLAyJavJ0sX4t4REuJIlYAVx9g9ihOEQ3fPfO0i-DflVnCciZkTigGpRAfMQSzdKDbuA0cvV2eJPUNtM4FCujVZqgBgTUJZcl-Bujdi0pGpxC7_amRhAuB7n&sa=X&sqi=2&ved=2ahUKEwir0pjupeSFAxWMSFUIHefZCYoQxKsJegQIQhAB&ictx=0)
[Spanisch Deutsch](https://www.google.com/search?client=ms-android-motorola-rvo3&sca_esv=e4fcf10d584cbf3e&sca_upv=1&hl=de-AT&gl=at&cs=1&sxsrf=ACQVn0_l5Hgu-Vf
hallo, ich möchte fragen, ob es jemandem hallo ist es schon eingefallen, "Json-Code für Arduino" zu schreiben?
in "open-meteo app" sind die Wetter-Daten besser, als in "OWM" (vor allem "min/max Temp")

Aber!!!? Fehlt "Description", es gibt nur einen "Code".

Wie unter Berwendung von "Json-Daten" das Wetter mit Symbol im Arduino schreiben?

"if (weathercode = "0") {
    wetter = klar;
    Serial.printl(wetter);
   display.print("klar");
}

funktioniert nicht

Hat da jemand schon etwas in C für ESP32 in Arduino-IDE geschrieben?
hello, I would like to ask if anyone has thought of writing "Json code for Arduino"?
In "open-meteo app" the weather data is better than in "OWM" (especially "min/max temp")

But!!!? There is no “Description”, there is only a “Code”.

How to write weather with symbol in Arduino using "Json data"?

"if (weathercode = "0") {
    weather = clear;
    Serial.printl(weather);
   display.print("clear");
}

doesn't work

Has anyone already written something in C for ESP32 in the Arduino IDE?

**Comment by tmabell** [comment]
Patrick,
 
You have done some remarkable things lately!  I use your data all over my website.  I do wish I could fill in the multiple “n/a” on this page however…
 
https://www.mymishawakaweather.com/4CastCompareMODELS.php
 
Thanks for all you do!
 
Tom

**Reply by patrick-zippenfenig** [reply]
All models now calculate precipitation probability. See here: https://open-meteo.com/en/docs#hourly=precipitation_probability&models=ecmwf_ifs025,bom_access_global,gfs_seamless,icon_seamless,gem_seamless

NBM is not yet available, but I might add it in the near future

**Comment by tmabell** [comment]
Thanks!  I had been checking back there periodically but kind of got away from it.  This is VERY helpful!  Thank you again.
 
Tom

**Comment by tmabell** [comment]
Patrick,

 �

I have updated my page as well as added the UKMET but I noticed that while the UKMO has daily precipitation probabilities, it lacks hourly values for precipitation probability.  Is this in the works?

 �

https://mymishawakaweather.com/4CastCompareMODELS.php 

 �

Thank you once again Patrick,

 �

Tom

 �

**Reply by patrick-zippenfenig** [reply]
UKMO models should not show any precipitation probabilities. UK MetOffice did not make their ensemble models open-data.

**Comment by tmabell** [comment]
Sorry, another quick question Patrick.  The UKMO Seamless output has an issue where the value for Day 6 “precipitation_sum_ukmo_seamless” is absent.  There is a value for “snowfall_sum_ukmo_seamless” on day 6 so I wondered if it was a bug.

 �

**Reply by patrick-zippenfenig** [reply]
UKMO only offers 7 days of forecast. Depending on the timezone and currently available data, day 6 may have some missing data for the last hours. The API is therefore not able to return daily data for day 6 due to partial missing data.
